arkouda/categorical.py

```diff
@@ -366,10 +366,59 @@ def endswith(self, substr : str) -> pdarray:
         categoriesendswith = self.categories.endswith(substr)
         return categoriesendswith[self.codes]
 
-    def in1d(self, test):
-        #__doc__ = in1d.__doc__
-        categoriesisin = in1d(self.categories, test)
-        return categoriesisin[self.codes]
+    @typechecked
+    def in1d(self, test : Union[Strings,Categorical]) -> pdarray:
+        """
+        Test whether each element of the Categorical.categories object is 
```

Comment:
This is not quite correct -- it's not testing whether each element of `Categorical.categories` is in the test set, but each element of the `Categorical` itself. The returned array should be the same size as the Categorical, not necessarily the same size as `Categorical.categorie`s.

The docstring should not mention `.categories`, because that is an implementation detail abstracted from the user. From the user's perspective, the `Categorical` is an array of strings (not categories), and this function is testing membership of each of those strings in a test set.

Reply:
@reuster986 ready for review